lib/trie/decode.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package trie
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/ChainSafe/gossamer/internal/trie/node"
+	"github.com/ChainSafe/gossamer/internal/trie/pools"
+)
+
+var (
+	ErrReadHeaderByte  = errors.New("cannot read header byte")
+	ErrUnknownNodeType = errors.New("unknown node type")
+)
+
+func decodeNode(reader io.Reader) (n Node, err error) {
```

Comment:
I think this function should be in the `node` package with public sig `Decode(reader) (n Node, err error)`.  That way all the `DecodeBranch|Leaf` functions can be private.

Reply:
Totally! Done in 4be79860a78972e191b9d9aa1cc7888f9876f7a5